pymc_marketing/mmm/evaluation.py

```diff
@@ -0,0 +1,535 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Evaluation and diagnostics for MMM models."""
+
+import arviz as az
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+from sklearn.metrics import (
+    mean_absolute_error,
+    mean_absolute_percentage_error,
+    root_mean_squared_error,
+)
+
+from pymc_marketing.mmm.mmm import MMM
+
+
+class MMMEvaluator:
+    """A class to evaluate and diagnose PyMC-Marketing MMMs.
+
+    Parameters
+    ----------
+    model : MMM
+        The PyMC-Marketing MMM object.
+    idata : az.InferenceData
+        The InferenceData object returned by the sampling method.
+
+    Raises
+    ------
+    ValueError
+        If the model object does not have a non-None 'idata' attribute.
+    ValueError
+        If the model object does not have a non-None 'model' attribute.
+
+    """
+
+    def __init__(self, model: MMM):
+        if not hasattr(model, "idata") or model.idata is None:
+            raise ValueError("The model object must have a non-None 'idata' attribute.")
+        if not hasattr(model, "model") or model.model is None:
+            raise ValueError("The model object must have a non-None 'model' attribute.")
+
+        self.model = model
+        self.metric_distributions: dict[str, np.ndarray] = {}
+        self.metric_summaries: dict[str, dict[str, float]] = {}
+
+    # Same error metric as Robyn
+    @staticmethod
+    def nrmse(y_true: np.ndarray, y_pred: np.ndarray) -> float:
```

Comment:
Shall we put these metrics in https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/metrics.py?